lsp-methods.el

```diff
@@ -711,10 +711,16 @@ interface Range {
   (inline-quote (plist-get (plist-get ,range :end) :line)))
 
 (defun lsp--apply-workspace-edits (edits)
-  (cl-letf (((lsp--workspace-change-timer-disabled lsp--cur-workspace) t))
-    (maphash (lambda (key value)
-               (lsp--apply-workspace-edit key value))
-             (gethash "changes" edits))))
+  (cl-letf (((lsp--workspace-change-timer-disabled lsp--cur-workspace) t)
+             (fn (lambda (key value)
+                   (lsp--apply-workspace-edit key value))))
```

Comment:
`documentChanges` is actually an array of `TextDocumentEdit` objects, which use `TextEdit` instead of `WorkspaceEdit`, so you would need to use `lsp--apply-text-edits` instead.